pkg/appstatus/client.go

```diff
@@ -0,0 +1,91 @@
+package appstatus
+
+import (
+	"context"
+	"io"
+	"os/exec"
+	"time"
+
+	"github.com/giantswarm/microerror"
+	"github.com/sirupsen/logrus"
+)
+
+type Config struct {
+	Logger *logrus.Logger
+	Stderr io.Writer
+}
+
+type Client struct {
+	logger *logrus.Logger
+	stderr io.Writer
+}
+
+func New(config Config) (*Client, error) {
+	if config.Logger == nil {
+		return nil, microerror.Maskf(invalidConfigError, "%T.Logger must not be empty", config)
+	}
+	if config.Stderr == nil {
+		return nil, microerror.Maskf(invalidConfigError, "%T.Stderr must not be empty", config)
+	}
+
+	c := &Client{
+		logger: config.Logger,
+		stderr: config.Stderr,
+	}
+
+	return c, nil
+}
+
+func (c *Client) WaitForAppDeployment(ctx context.Context, appName, orgNamespace, managementCluster string, timeout time.Duration) error {
+	c.logger.Infof("Waiting for app %s to be deployed in namespace %s", appName, orgNamespace)
+
+	// Login to management cluster
+	loginCmd := exec.Command("tsh", "kube", "login", managementCluster)
```

Comment:
I'm not sure I like tools to play with `kubectl` state like that, but could find a better way after reading `tsh kube login` docs:
```suggestion
	loginCmd := exec.Command("tsh", "kube", "login", managementCluster)
```